doc/en/api/1.0.0/resource.yaml

```diff
@@ -0,0 +1,210 @@
+---
+swagger: '2.0'
+info:
+  version: 1.0.0
+  title: GeoServer Resources
+  description: A resource is any item in the data directory that does not represent configuration. Typical resources include styles and icons.
+
+paths:
+  /resource/{pathToResource}:
+    parameters:
+      - name: pathToResource
+        in: path
+        type: string
+        required: true
+        description: The full path to the resource. Required, but may be empty; a request to `resource` references the top level resource directory.
+    get:
+      operationId: resourceGet
+      description: Download a resource, list contents of directory, or show formatted resource metadata.
+      produces: 
+        - application/xml
+        - application/json
+        - text/plain
+        - "*/*"
+      parameters:
+        - name: operation
+          in: query
+          type: string
+          description: The type of GET operation. `default` returns a list of the contained resources in the case of a directory resource, or the actual resource contents in the case of a resource resource.`metadata` requests a metadata summary of the resource.
+          required: false
+          default: default
+          enum:
+            - default
+            - metadata
+            - move
+            - copy
+        - name: format
+          in: query
+          type: string
+          description: The format of the response. Only applicable for the `metadata` operation, or for a directory resource.
+          required: false
+          default: html
+          enum:
+            - html
+            - xml
+            - json
+      responses:
+        200:
+          description: OK
+          headers:
```

Comment:
while we document the header here, do we also need to document the response body?

Reply:
We should, yes. Unfortunately, that is impossible within the constraints of the Swagger 2.0 API, due to the variety in possible bodies. I tried a bunch of approached within the constraints of the API, but didn't get anything that actually worked.

I think I'm going to just add a bunch of examples to the description

Reply:
Could we document the metadata response (the only structured response) and use the description to document the `*/*` case?

Reply:
Its a bit more complicated than that unfortunately.
A regular GET against a resource will return the actual resource
A regular GET against a directory will return a "ResourceDirectory", which is a structured response listing information about the directory and its children.
A GET with operation=METADATA will return a "ResourceMetadata", which is a structured response listing information about the resource. This will be returned even if the resource is a directory, and does not list the children.

I have schemas for both these responses in the definitions section, but wasn't able to construct a suitable response definition for all the possible cases.